src/service-worker/install.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {calculateScriptBaseUrl} from '../service/extensions-impl';
+import {isExperimentOn} from '../experiments';
+import {dev} from '../log';
+
+/** @const */
+const TAG = 'cache-service-worker';
+
+/**
+ * Registers the Google AMP Cache service worker if the browser supports SWs.
+ */
+export function installCacheServiceWorker(win) {
+  if (isExperimentOn(win, TAG) && 'serviceWorker' in navigator) {
+    const base = calculateScriptBaseUrl(win.location.pathname,
+      getMode().isLocalDev, getMode().test);
+    const url = `${base}/sw.js`;
+    navigator.serviceWorker.register(url).then(reg => {
```

Comment:
I commented on this before. We should not.

On Wed, Jul 20, 2016 at 10:28 AM, Dima Voytenko <notifications@github.com>
wrote:

> In src/service-worker/install.js
> <https://github.com/ampproject/amphtml/pull/4037#discussion_r71569506>:
>
> > +import {calculateScriptBaseUrl} from '../service/extensions-impl';
> > +import {isExperimentOn} from '../experiments';
> > +import {dev} from '../log';
> > +
> > +/** @const */
> > +const TAG = 'cache-service-worker';
> > +
> > +/**
> > + * Registers the Google AMP Cache service worker if the browser supports SWs.
> > + */
> > +export function installCacheServiceWorker(win) {
> > +  if (isExperimentOn(win, TAG) && 'serviceWorker' in navigator) {
> > +    const base = calculateScriptBaseUrl(win.location.pathname,
> > +      getMode().isLocalDev, getMode().test);
> > +    const url = `${base}/sw.js`;
> > +    navigator.serviceWorker.register(url).then(reg => {
>
> Do we do this even for documents NOT served via cdn.ampproject.org?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ampproject/amphtml/pull/4037/files/03466c7810dac4594c51c7af988b008f9eda456a#r71569506>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAFeT58gXlesyOewF6Q3qQaPNqjRlCoJks5qXlq9gaJpZM4JK6Ck>
> .
>